llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -625,14 +634,34 @@ void AArch64AsmPrinter::emitHwasanMemaccessSymbols(Module &M) {
                                      .addImm(4)
                                      .addImm(55),
                                  *STI);
-    OutStreamer->emitInstruction(
-        MCInstBuilder(AArch64::LDRBBroX)
-            .addReg(AArch64::W16)
-            .addReg(IsShort ? AArch64::X20 : AArch64::X9)
-            .addReg(AArch64::X16)
-            .addImm(0)
-            .addImm(0),
-        *STI);
+
+    if (IsFixedShadow) {
+      std::optional<uint64_t> HwasanFixedShadowBase = getFixedShadowBase();
+
+      OutStreamer->emitInstruction(
+          MCInstBuilder(AArch64::MOVZXi)
+              .addReg(AArch64::X17)
+              .addImm(HwasanFixedShadowBase.value_or(0) >> 32)
```

Comment:
I don't understand `value_or(0)`. Shouldn't we crash instead?

Reply:
report_fatal_error >> value_or(0) >> assert()

Reply:
Oh, proabable a better alternative is to 
pass shadow as ImmArg<ArgIndex<>> and 
generate callback like __hwasan_check_x0_2_short_fixed_123123123_v2,
?

Reply:
Done!